webapp/lib/app/nook/view.dart

```diff
@@ -413,34 +426,93 @@ class ConversationPanelView with AutomaticSuggestionIndicator {
   }
 }
 
+class DateSeparatorView {
+  DateTime _dateTime;
+  
+  DivElement renderElement;
+  SpanElement _dateElement;
+
+  String get formattedDateString => _getDaysAgoSinceToday(_dateTime);
+  set dateTime (DateTime dateTime) {
+    this._dateTime = dateTime;
+    _dateElement.innerText = formattedDateString;
+  }
+
+  DateSeparatorView(this._dateTime) {
+    renderElement = DivElement()..className = 'messages-date-separator__wrapper';
+    _dateElement = SpanElement()
+      ..className = 'messages-date-separator'
+      ..innerText = formattedDateString;
+    renderElement.append(_dateElement);
+  }
+
+  void hide() {
+    renderElement.setAttribute('hidden', 'true');
+  }
+
+  void show() {
+    renderElement.removeAttribute('hidden');
+  }
+
+  String _getDaysAgoSinceToday(DateTime dateTime) {
+    final date = DateTime(dateTime.year, dateTime.month, dateTime.day);
+
+    final today = DateTime.now();
+    final todate = DateTime(today.year, today.month, today.day + 1);
+
+    final differenceDays = todate.difference(date).inDays;
+    final DateFormat formatter = DateFormat('MMM yyyy');
+
+    if (differenceDays <= 1) {
+      return "Today";
+    } else if (differenceDays <= 2) {
+      return "Yesterday";
+    } else if (differenceDays <= 7) {
+      return "This week";
+    } else if (differenceDays > 7) {
+      return formatter.format(dateTime);
+    }
```

Comment:
the week-related calculation isn't working right, because the start of a week is relative to the current day, rather than just being 7 days ago. What we'd need to do is to compute the week number as per [this stackoverflow answer](https://stackoverflow.com/a/54129275), and then compare if they're equal. We can land this PR without it for now, and add this functionality in a follow up, whichever you prefer.

below is the code a bit simplified & comparisons adjusted so they don't need the `+1` from the previous line.
```suggestion
    if (differenceDays < 1) return "Today";
    if (differenceDays == 1) return "Yesterday";
    return formatter.format(dateTime);
```